File: chrome/test/data/webui/new_tab_page/new_tab_page_browsertest.js
// Copyright 2019 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/** @fileoverview Test suite for the WebUI new tab page page. */

GEN_INCLUDE(['//chrome/test/data/webui/polymer_browser_test_base.js']);

GEN('#include "build/build_config.h"');
GEN('#include "build/chromeos_buildflags.h"');
GEN('#include "components/search/ntp_features.h"');
GEN('#include "content/public/test/browser_test.h"');

class NewTabPageBrowserTest extends PolymerTest {
  /** @override */
  get browsePreload() {
    throw new Error('this is abstract and should be overriden by subclasses');
  }
}

var NewTabPageModulesHistoryClustersModuleTest =
    class extends NewTabPageBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=new_tab_page/modules/history_clusters/module_test.js';
  }

  /** @override */
  get featureList() {
    return {
      enabled: [
        'ntp_features::kNtpHistoryClustersModule',
      ],
    };
  }
};

TEST_F('NewTabPageModulesHistoryClustersModuleTest', 'Core', function() {
  runMochaSuite('NewTabPageModulesHistoryClustersModuleTest core');
});

TEST_F('NewTabPageModulesHistoryClustersModuleTest', 'Layouts', function() {
  runMochaSuite('NewTabPageModulesHistoryClustersModuleTest layouts');
});

TEST_F(
    'NewTabPageModulesHistoryClustersModuleTest',
    'UnloadMetricImageDisplayStateNone', function() {
      runMochaSuite(
          'NewTabPageModulesHistoryClustersModuleTest unload metric no images');
    });

TEST_F(
    'NewTabPageModulesHistoryClustersModuleTest',
    'UnloadMetricImageDisplayStateAll', function() {
      runMochaSuite(
          'NewTabPageModulesHistoryClustersModuleTest unload metric all images');
    });

TEST_F(
    'NewTabPageModulesHistoryClustersModuleTest', 'CartTileRendering',
    function() {
      runMochaSuite(
          'NewTabPageModulesHistoryClustersModuleTest cart tile rendering');
    });

var NewTabPageModulesHistoryClustersModuleTileTest =
    class extends NewTabPageBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=new_tab_page/modules/history_clusters/tile_test.js';
  }

  /** @override */
  get featureList() {
    return {
      enabled: [
        'ntp_features::kNtpHistoryClustersModule',
      ],
    };
  }
};

TEST_F('NewTabPageModulesHistoryClustersModuleTileTest', 'All', function() {
  mocha.run();
});

var NewTabPageModulesHistoryClustersModuleSuggestTileTest =
    class extends NewTabPageBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=new_tab_page/modules/history_clusters/suggest_tile_test.js';
  }

  /** @override */
  get featureList() {
    return {
      enabled: [
        'ntp_features::kNtpHistoryClustersModule',
      ],
    };
  }
};

TEST_F(
    'NewTabPageModulesHistoryClustersModuleSuggestTileTest', 'All', function() {
      mocha.run();
    });

var NewTabPageModulesHistoryClustersModuleCartTileTest =
    class extends NewTabPageBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=new_tab_page/modules/history_clusters/cart/cart_tile_test.js';
  }
};

TEST_F('NewTabPageModulesHistoryClustersModuleCartTileTest', 'All', function() {
  mocha.run();
});

var NewTabPageModulesHistoryClustersV2ModuleTest =
    class extends NewTabPageBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=new_tab_page/modules/v2/history_clusters/module_test.js';
  }

  /** @override */
  get featureList() {
    return {
      enabled: [
        'ntp_features::kNtpHistoryClustersModule',
        'ntp_features::kNtpModulesRedesigned',
      ],
    };
  }
};

TEST_F('NewTabPageModulesHistoryClustersV2ModuleTest', 'Core', function() {
  runMochaSuite('NewTabPageModulesHistoryClustersV2ModuleTest core');
});

GEN('#if !defined(OFFICIAL_BUILD)');

var NewTabPageModulesPhotosModuleTest = class extends NewTabPageBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=new_tab_page/modules/photos/module_test.js';
  }
};

TEST_F('NewTabPageModulesPhotosModuleTest', 'All', function() {
  mocha.run();
});

GEN('#endif  // !defined(OFFICIAL_BUILD)');

var NewTabPageDiscountConsentDialogTest = class extends NewTabPageBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=new_tab_page/modules/cart/discount_consent_dialog_test.js';
  }
};

TEST_F('NewTabPageDiscountConsentDialogTest', 'All', function() {
  mocha.run();
});

var NewTabPageDiscountConsentCartTest = class extends NewTabPageBrowserTest {
  /** @override */
  get browsePreload() {
    return 'chrome://new-tab-page/test_loader.html?module=new_tab_page/modules/cart/discount_consent_card_test.js';
  }
};

TEST_F('NewTabPageDiscountConsentCartTest', 'All', function() {
  mocha.run();
});

File: chrome/test/data/webui/new_tab_page/new_tab_page_browsertest.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/test/scoped_feature_list.h"
#include "build/build_config.h"
#include "chrome/common/webui_url_constants.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "components/history_clusters/core/features.h"
#include "components/search/ntp_features.h"
#include "content/public/test/browser_test.h"

class NewTabPageBrowserTest : public WebUIMochaBrowserTest {
 protected:
  NewTabPageBrowserTest() {
    set_test_loader_host(chrome::kChromeUINewTabPageHost);
  }
};

using NewTabPageTest = NewTabPageBrowserTest;

IN_PROC_BROWSER_TEST_F(NewTabPageTest, MetricsUtils) {
  RunTest("new_tab_page/metrics_utils_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageTest, VoiceSearchOverlay) {
  RunTest("new_tab_page/voice_search_overlay_test.js", "mocha.run()");
}

// TODO(crbug.com/40933410):  Re-enable once no longer fails.
IN_PROC_BROWSER_TEST_F(NewTabPageTest, DISABLED_LensForm) {
  RunTest("new_tab_page/lens_form_test.js", "mocha.run()");
}

// TODO(crbug.com/40902230): Test is flaky across platforms.
IN_PROC_BROWSER_TEST_F(NewTabPageTest, DISABLED_LensUploadDialog) {
  RunTest("new_tab_page/lens_upload_dialog_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageTest, Logo) {
  RunTest("new_tab_page/logo_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageTest, DoodleShareDialog) {
  RunTest("new_tab_page/doodle_share_dialog_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageTest, BackgroundManager) {
  RunTest("new_tab_page/background_manager_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageTest, MiddleSlotPromo) {
  RunTest("new_tab_page/middle_slot_promo_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageTest, MobilePromo) {
  RunTest("new_tab_page/mobile_promo_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageTest, ImageProcessor) {
  RunTest("new_tab_page/image_processor_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageTest, Transparency) {
  RunTest("new_tab_page/transparency_test.js", "mocha.run()");
}

using NewTabPageModulesTest = NewTabPageBrowserTest;

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, ModuleWrapper) {
  RunTest("new_tab_page/modules/module_wrapper_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, ModulesV2) {
  RunTest("new_tab_page/modules/v2/modules_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, ModuleHeaderV2) {
  RunTest("new_tab_page/modules/v2/module_header_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, Modules) {
  RunTest("new_tab_page/modules/modules_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, ModuleDescriptor) {
  RunTest("new_tab_page/modules/module_descriptor_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, ModuleRegistry) {
  RunTest("new_tab_page/modules/module_registry_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, ModuleHeader) {
  RunTest("new_tab_page/modules/module_header_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, InfoDialog) {
  RunTest("new_tab_page/modules/info_dialog_test.js", "mocha.run()");
}

#if !defined(OFFICIAL_BUILD)
// The dummy module is not available in official builds.
IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, DummyModule) {
  RunTest("new_tab_page/modules/v2/dummy/module_test.js", "mocha.run()");
}
#endif  // !defined(OFFICIAL_BUILD)

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, Calendar) {
  RunTest("new_tab_page/modules/v2/calendar/calendar_test.js",
          "runMochaSuite('NewTabPageModulesCalendarTest general')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, CalendarMetrics) {
  RunTest("new_tab_page/modules/v2/calendar/calendar_test.js",
          "runMochaSuite('NewTabPageModulesCalendarTest metrics')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, CalendarEvent) {
  RunTest("new_tab_page/modules/v2/calendar/calendar_event_test.js",
          "runMochaSuite('NewTabPageModulesCalendarEventTest general')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, CalendarEventMetrics) {
  RunTest("new_tab_page/modules/v2/calendar/calendar_event_test.js",
          "runMochaSuite('NewTabPageModulesCalendarEventTest metrics')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, GoogleCalendarModule) {
  RunTest("new_tab_page/modules/v2/calendar/google_calendar_module_test.js",
          "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, OutlookCalendarModule) {
  RunTest("new_tab_page/modules/v2/calendar/outlook_calendar_module_test.js",
          "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, DriveModule) {
  RunTest("new_tab_page/modules/drive/module_test.js", "mocha.run()");
}

IN_PROC_BROWSER_TEST_F(NewTabPageModulesTest, FileSuggestionModule) {
  RunTest("new_tab_page/modules/v2/file_suggestion/module_test.js",
          "mocha.run()");
}

using NewTabPageAppTest = NewTabPageBrowserTest;

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, Misc) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest Misc')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, OgbThemingRemoveScrim) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest OgbThemingRemoveScrim')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, OgbScrim) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest OgbScrim')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, Theming) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest Theming')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, Promo) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest Promo')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, Clicks) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest Clicks')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, Modules) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest Modules')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, V2Modules) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest V2Modules')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, CounterfactualModules) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest CounterfactualModules')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, CustomizeChromeSidePanel) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest CustomizeChromeSidePanel')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, LensUploadDialog) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest LensUploadDialog')");
}

IN_PROC_BROWSER_TEST_F(NewTabPageAppTest, WallpaperSearch) {
  RunTest("new_tab_page/app_test.js",
          "runMochaSuite('NewTabPageAppTest WallpaperSearch')");
}

class NewTabPageModulesMostRelevantTabResumptionModuleTest
    : public NewTabPageBrowserTest {
 protected:
  NewTabPageModulesMostRelevantTabResumptionModuleTest() {
    scoped_feature_list_.InitWithFeatures(
        /*enabled_features=*/{ntp_features::
                                  kNtpMostRelevantTabResumptionModule},
        /*disabled_features=*/{});
  }

 private:
  base::test::ScopedFeatureList scoped_feature_list_;
};

IN_PROC_BROWSER_TEST_F(NewTabPageModulesMostRelevantTabResumptionModuleTest,
                       Core) {
  RunTest("new_tab_page/modules/v2/most_relevant_tab_resumption/module_test.js",
          "runMochaSuite('NewTabPageModulesMostRelevantTabResumptionModuleTest "
          "Core')");
}
